src/yearn.ts

```diff
@@ -118,10 +125,15 @@ export class Yearn<T extends ChainId> {
     const allowListService = allowlistAddress
       ? new AllowListService(chainId, this.context, allowlistAddress)
       : undefined;
+    const partnerServiceAddress = this.context.partnerId && PartnerService.addressByChain(chainId);
+    const partnerService = partnerServiceAddress
+      ? new PartnerService(chainId, this.context, partnerServiceAddress)
+      : undefined;
```

Comment:
nit: I  think the ternary statement here could be a bit more readable if we refactor to something along the lines:

```typescript
const partnerService = this.getPartnerService({
  address: partnerServiceAddress,
  chainId
});

// ...

private getPartnerService({ address, chainId }: { address: string | null; chainId: ChainId }) {
  if (!address) return;

  return new AllowListService(chainId, this.context, address);
}
```

We should also try and avoid not having named params, it does make the code more verbose but much more flexible because one does not need to pass the arguments in any order.

Reply:
I agree on the params stuff (btw the ones that we have right now are called positional params, while what you are suggesting are actually called named params since you actually have to specify the name 😄  ).

On the ternary I'm not sure I 100% agree, what we have right now easily allows you to see the condition on when the services are going to be initalized or not. Moving all that into a new function, while it might make it less code to read initially, can actually obfuscate what getPartnerService is going to do. In this way we clearly and rapidly know when the service is getting initialized or not. What do you think?

Reply:
> I agree on the params stuff (btw the ones that we have right now are called positional params, while what you are suggesting are actually called named params since you actually have to specify the name 😄 ).

Correct, that's why I said "avoid not having named params" :) maybe you've missed the "not"

With regards to the ternary, I don't know, I think it might be ok for now, but when if the logic start compounding then it becomes messy. I've started to see that in some places, where then we have nested ternary statements. I reckon `undefined` should be used when something is not defined, and not set explicitly... I still prefer the function :)